### PR TITLE
DLS-6750 Refactor error handling

### PR DESCRIPTION
**Pull request by LAJW**
This service doesn't log any PII, though I thought it would be a good opportunity to rework some of the error handling, which tends to be pretty ugly and makes it difficult to diagnose what is and isn't logged.

### CODE REVIEW

**Comment by johno1985** [comment]
PR build failed

**Review by TomRafferty — APPROVED** [review]
Much cleaner! Nice work 😄

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/actions/AuthenticatedActionWithRetrievedData.scala [hunk 1]
```diff
@@ -145,7 +131,7 @@ class AuthenticatedActionWithRetrievedData @Inject() (
               )
             )
 
-          case (Right(AffinityGroup.Individual), cl, Some(nino)) if cl >= ConfidenceLevel.L200 =>
```
Comment (johno1985) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/actions/AuthenticatedActionWithRetrievedData.scala:150]: does removing the cl check change the behaviour?

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/util/HttpResponseOps.scala [hunk 2]
```diff
@@ -29,21 +29,21 @@ object HttpResponseOps {
...
-            case Some(path) => Right(response.json \ path)
-          } catch {
-            case error: Throwable => Left(s"Could not read http response as JSON: ${error.getMessage}")
+        jsLookupResult =
+          path match {
+            case None       => JsDefined(json)
+            case Some(path) => json \ path
           }
         // use Option here to filter out null values
-        result         <- jsLookupResult.toOption.toRight("No JSON found in body of http response")
```
Comment (johno1985) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/util/HttpResponseOps.scala]: is it possible that the value of jsLookupResult be a Left?

**Review by johno1985 — COMMENTED** [review]
minor comments/questions

**Review by johno1985 — APPROVED** [review]

**Pull request merged**